packages/flutter/lib/src/material/search.dart

```diff
@@ -275,6 +276,11 @@ abstract class SearchDelegate<T> {
   /// Defaults to [TextInputAction.search].
   final TextInputAction textInputAction;
 
+  /// Sets the maintainState property on the delegate's PageRoute.
```

Comment:
I think this can be a little more descriptive. You can just grab the same description from [ModalRoute.maintainState](https://master-api.flutter.dev/flutter/widgets/ModalRoute/maintainState.html).

Reply:
+1 this should explain a little more what setting this to true means in the context of the search page. The fact that the search is pushing a PageRoute is an implementation detail, that should not be exposed here.